doc/book/admin/disaster_recovery.rst

```diff
@@ -1,126 +1,162 @@
-   :ref:`box.cfg.read_only <cfg_basic-read_only>` parameter to *false* and let
-   the load be handled by the replica (effective master).
+**Actions:**
 
-3. Set up a replacement for the crashed master on a spare host, with
-   :ref:`replication <cfg_replication-replication>` parameter set to replica
-   (effective master), so it begins to catch up with the new master’s state.
-   The new instance should have :ref:`box.cfg.read_only <cfg_basic-read_only>`
-   parameter set to *true*.
+1.  Ensure the master is stopped.
+    For example, log in to the master machine and use ``tt stop``.
 
-You lose the few transactions in the master
-:ref:`write ahead log file <index-box_persistence>`, which it may have not
-transferred to the replica before crash. If you were able to salvage the master
-.xlog file, you may be able to recover these. In order to do it:
+2.  Configure a new replica set leader using the :ref:`<replicaset_name>.leader <configuration_reference_replicasets_name_leader>` option.
 
-1. Find out the position of the crashed master, as reflected on the new master.
+3.  Reload configuration on all instances using :ref:`config:reload() <config-module>`.
 
-   a. Find out instance UUID from the crashed master :ref:`xlog <internals-wal>`:
+4.  Make sure that a new replica set leader is a master using :ref:`box.info.ro <box_introspection-box_info>`.
 
-      .. code-block:: console
+5.  On a new master, :ref:`remove a crashed instance from the '_cluster' space <replication-remove_instances-remove_cluster>`.
 
-          $ head -5 *.xlog | grep Instance
-          Instance: ed607cad-8b6d-48d8-ba0b-dae371b79155
+6.  Start the instance again on a spare host.
 
-   b. On the new master, use the UUID to find the position:
+See also: :ref:`Performing manual failover <replication-controlled_failover>`.
 
-      .. code-block:: tarantoolsession
 
-          tarantool> box.info.vclock[box.space._cluster.index.uuid:select{'ed607cad-8b6d-48d8-ba0b-dae371b79155'}[1][1]]
-          ---
-          - 23425
-          <...>
+.. _admin-disaster_recovery-master_replica_auto_failover:
 
-2. Play the records from the crashed .xlog to the new master, starting from the
-   new master position:
+Master crash: automated failover
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   a. Issue this request locally at the new master's machine to find out
-      instance ID of the new master:
+**Configuration:** master-replica (:ref:`automated failover <replication-bootstrap-auto>`).
 
-      .. code-block:: tarantoolsession
+**Problem:** The master has crashed.
 
-          tarantool> box.space._cluster:select{}
-          ---
-          - - [1, '88580b5c-4474-43ab-bd2b-2409a9af80d2']
-          ...
+**Actions:**
 
-   b. Play the records to the new master:
+1.  Use ``box.info.election`` to make sure a new master is elected automatically.
 
-      .. code-block:: console
+2.  Remove a crashed master from a replica set.
+
+3.  Set up a replacement for the crashed master on a spare host.
```

Comment:
Sorry, I'm a bit confused :) Please help clarify this question. Here is the previous comment:

> This only makes sense when the user moves the master from one physical node to another. But even then, with the instance names, the master on a new physical node will occupy the same _cluster slot (it will find it by name).

Don't know what the author of the initial topic kept in mind but let's assume that the old master goes down and we are setting up a replacement for the old master on a new host (as you mentioned above).

Reply:
Yes, we should remove the old host from _cluster in this case. Let's not change the logic of the previous author.